core/src/main/java/org/infinispan/topology/CacheTopologyControlCommand.java

```diff
@@ -36,8 +36,8 @@
       JOIN,
       // A member is signaling that it wants to leave the cluster.
       LEAVE,
-      // A member is confirming that it finished the rebalance operation.
-      REBALANCE_CONFIRM,
+      // A member is confirming that it has finished the topology change during rebalance.
+      REBALANCE_PHASE_CONFIRM,
```

Comment:
1. You don't request/receive new state, but IIRC you block sending another topology until you receive the confirmation of installment from all live nodes. The member confirms that exactly what you describe happened.
2. ok, I admit I use the first article that comes to my mind if at all :)